core/trino-main/src/main/java/io/trino/execution/scheduler/ScaledWriterScheduler.java

```diff
@@ -95,12 +98,19 @@ private int getNewTaskCount()
                 .mapToDouble(full -> full ? 1.0 : 0.0)
                 .average().orElse(0.0);
 
-        long writtenBytes = writerTasksProvider.get().stream()
-                .map(TaskStatus::getPhysicalWrittenDataSize)
-                .mapToLong(DataSize::toBytes)
-                .sum();
+        boolean taskWritersLimitExceeded;
+        if (taskScaleWriters) {
+            taskWritersLimitExceeded = writerTasksProvider.get().stream().allMatch(TaskStatus::isWritersFullyScaled);
+        }
+        else {
+            long writtenBytes = writerTasksProvider.get().stream()
+                    .map(TaskStatus::getPhysicalWrittenDataSize)
+                    .mapToLong(DataSize::toBytes)
+                    .sum();
+            taskWritersLimitExceeded = (writtenBytes >= (writerMinSizeBytes * scheduledNodes.size()));
```

Comment:
Looking at this it seems we previously weren't waiting for new task to write `min size bytes`. According to this formula all workers in total had to write `writerMinSizeBytes` data (because `getPhysicalWrittenDataSize` is cumulative)

The condition for `taskScaleWriters` won't scale if there is a single task that is not fully scalled. This becomes a potential chokepoint in a distributed system.

Maybe we simple modify the condition here to be:
```
(writtenBytes >= (writerMinSizeBytes * scheduledNodes.size() * maxWriterCount));
```
It seems `writerCount` wasn't taken into account previously anyway.

Probably a new task should start with already writer threads maxed out